File: PMDS.GUI.VB/planArchive2/plan/frmPdfViewer.vb



Public Class frmPdfViewer



    Private Sub frmPdfViewer_Load(sender As Object, e As EventArgs) Handles MyBase.Load

    End Sub

    Public Sub initControl(sFileName As String, fileStream As System.IO.Stream)
        Try
            Me.Text = doUI.getRes("PdfViewer")
            Me.Icon = QS2.Resources.getRes.getIcon(QS2.Resources.getRes.Launcher.ico_PMDS, 32, 32)

            If fileStream Is Nothing Then
                Me.PdfViewerControl1.Load(sFileName, "")
            Else
                Me.PdfViewerControl1.Load(fileStream)
            End If

            Me.PdfViewerControl1.ShowVerticalScrollBar = True
            Me.PdfViewerControl1.ZoomMode = Syncfusion.Windows.Forms.PdfViewer.ZoomMode.FitWidth

        Catch ex As Exception
            Throw New Exception("frmPdfViewer.initControl: " + ex.ToString())
        End Try
    End Sub

End Class

File: QS2.Design.Auto/print/frmPDFViewer.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using QS2.Resources;


namespace qs2.design.auto.print
{
    public partial class frmPDFViewer : Form
    {
        public frmPDFViewer()
        {
            InitializeComponent();
        }

        private void frmPDFViewer_Load(object sender, EventArgs e)
        {
            Icon = getRes.getIcon(getRes.Allgemein2.ico_PDF, 32, 32);

            if (!qs2.core.license.doLicense.GetLicense(qs2.core.Enums.eLicense.LIC_DOCUMENTS).bValue)
                Text = qs2.core.language.sqlLanguage.getRes("TrialVersion") + qs2.core.language.sqlLanguage.getRes("ValidTo") + " " + qs2.core.license.doLicense.GetLicense(qs2.core.Enums.eLicense.LIC_DOCUMENTS_VALID_DATE).dValue.ToString("dd.MM.yyyy");
            else
                Text = qs2.core.language.sqlLanguage.getRes("frmPDFViewer", "ALL", "ALL", false, true);
        }

        public bool Init(string rpt, qs2.ui.print.infoQry InfoQuery, qs2.ui.print.infoReport InfoReport)
        {
            contPDFViewer1.SetToolstrips(false, false, true, true);
            contPDFViewer1.Init(rpt, InfoQuery, InfoReport);
            return true;
        }
    }
}
